datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVAnalyzerRule.scala

```diff
@@ -83,8 +83,16 @@ class MVAnalyzerRule(sparkSession: SparkSession) extends Rule[LogicalPlan] {
         }
         Aggregate(grp, aExp, child)
     }
-    val catalog = DataMapStoreManager.getInstance().getDataMapCatalog(dataMapProvider,
+    var catalog = DataMapStoreManager.getInstance().getDataMapCatalog(dataMapProvider,
       DataMapClassProvider.MV.getShortName).asInstanceOf[SummaryDatasetCatalog]
+    // when first time DataMapCatalogs are initialized, it stores session info also, but when carbon
+    // session is newly created, catalog map will not be cleared, so if session info is different,
+    // remove the entry from map.
+    if (catalog != null && !catalog.mvSession.sparkSession.equals(sparkSession)) {
```

Comment:
If in case needAnalysis is false, can we skip this part for the corresponding query?

Reply:
no, irrespective of mv or not, the datamap should be loaded in memory